master/pkg/model/experiment_config.go

```diff
@@ -274,3 +275,58 @@ type InternalConfig struct {
 type NativeConfig struct {
 	Command []string `json:"command"`
 }
+
+// DebugConfig represents which subsystems will be debugged in the harness.
+type DebugConfig struct {
+	RootLogLevel             *string `json:"root_log_level"`
+	StorageLogLevel          *string `json:"storage_log_level"`
+	HarnessLogLevel          *string `json:"harness_log_level"`
+	DebugAllWorkers          bool    `json:"debug_all_workers"`
+	HorovodVerbose           bool    `json:"horovod_verbose"`
+	NCCLDebug                *string `json:"nccl_debug"`
+	NCCLDebugSubsys          *string `json:"nccl_debug_subsys"`
+	ResourceProfilePeriodSec float32 `json:"resource_profile_period_sec"`
+	StackTracePeriodSec      float32 `json:"stack_trace_period_sec"`
+}
+
+var (
+	debugConfigInfo  = "INFO"
+	debugConfigDebug = "DEBUG"
+	// trueDebugConfig is the value of DebugConfig that we assign for `debug: true`.
+	trueDebugConfig = DebugConfig{
+		RootLogLevel:             &debugConfigDebug,
```

Comment:
No, I think this is exactly what we did.  Since we used the root logger directly, when we called `logging.setLevel(logging.DEBUG)` we were setting the log level for all third party libraries.

Reply:
If debut was off we set all workers to INFO: https://github.com/determined-ai/determined/blob/68bdba57d782b094e740ce4b1cf5a739a18e1e38/harness/determined/exec/worker_process.py#L10

and then set it Warning for non-chief GPUs: https://github.com/determined-ai/determined/blob/68bdba57d782b094e740ce4b1cf5a739a18e1e38/harness/determined/_trial_controller.py#L274

Reply:
as we spoke about off-thread, this is the handling of the `debug: true` case, not the default case.